### PR TITLE
httpinstance.disable_system_trust: Don't fail if module 'Root Certs' …

### PR DESCRIPTION
**Pull request by dkupka**
…is not available

Server installation failed when attmpting to disable module 'Root Certs' and
the module was not available in HTTP_ALIAS_DIR. When the module is not
available there's no need to disable it and the error may be treated as
success.

https://pagure.io/freeipa/issue/6803

### CODE REVIEW

File: ipaserver/install/httpinstance.py [hunk 1]
```diff
@@ -355,9 +355,15 @@ def disable_system_trust(self):
...
 
-        result = ipautil.run(args + ['-list', name],
-                             env={},
-                             capture_output=True)
+        try:
+            result = ipautil.run(args + ['-list', name],
+                                 env={},
+                                 capture_output=True)
+        except ipautil.CalledProcessError:
+            root_logger.debug(
```
Comment (tiran) [inline ipaserver/install/httpinstance.py]: This catches and ignores all call error. I would rather limit the exception to exit code 29:

```
$ mkdir /tmp/nssdb
$ certutil -N -d /tmp/nssdb/ --empty-password
$ modutil -dbdir /tmp/nssdb/ -force -list 'Root Certs'
ERROR: Module "Root Certs" not found in database.
$ echo $?
29
```

**Comment by stlaz** [comment]
Hm, I believe the `-list` operation was there just to check whether the module is there. If `modutil` fails like this no matter the situation, it'd be better to just try to disable it whatever happens and go away with "meh, so what" if the disable fails.

**Comment by stlaz** [comment]
For the record:
```bash
[slaznick@vm-066 ~]$ sudo modutil -dbdir nssdb/ -disable 'Root Certs' -force
ERROR: Module "Root Certs" not found in database.
[slaznick@machine ~]$ echo $?
29
[slaznick@machine ~]$ 
```

**Comment by tiran** [comment]
@stlaz The broad except also catches and ignores typos in the command line or missing ```modutil``` binary.

**Comment by stlaz** [comment]
@tiran I of course agree on narrowing the broad except down, my point is we should rather remove the whole `-list` part and just try to do `-disable`.

**Comment by dkupka** [comment]
@tiran @stlaz Makes sense. I will update it accordingly. Thanks for suggestions.

**Comment by HonzaCholasta** [comment]
@stlaz, you can't do just `-disable`, as that would break upgrade (note that the `-list` is there because `-disable` always reports success).

**Comment by stlaz** [comment]
@HonzaCholasta You're right, I completely forgot about that one.

**Comment by stlaz** [comment]
This fixes the mentioned issue. I did not test whether the actual disable works but I should hope so as I don't see how this could break it.

**Comment by tomaskrizek** [comment]
ipa-4-5:

* 2a499551ca5ddf2596cc19a77f47c34e9f5c10c5 httpinstance.disable_system_trust: Don't fail if module 'Root Certs' is not available
master:

* 0128e805e591bc8ca5cea99739ad4cd7478df0b4 httpinstance.disable_system_trust: Don't fail if module 'Root Certs' is not available

**Pull request closed without merging**